File: code/generate-sharp/soup-tools/LoadBuildGraphResult.cs
// <copyright file="LoadBuildGraphResult.cs" company="Soup">
// Copyright (c) Soup. All rights reserved.
// </copyright>

using System.Collections.Generic;

namespace Soup.Tools;

public class LoadBuildGraph
{
	public int RootPackageGraphId { get; set; }
	public IList<PackageGraph> PackageGraphs { get; init; } = [];
	public IList<PackageInfo> Packages { get; init; } = [];
}

public class LoadBuildGraphResult
{
	public bool IsSuccess { get; set; }
	public string? Message { get; set; }
	public LoadBuildGraph? Graph { get; set; }
}

File: code/tools/view/load-build-graph-result.go
// <copyright file="load-build-graph-result.go" company="Soup">
// Copyright (c) Soup. All rights reserved.
// </copyright>

package main

type LoadBuildGraph struct {
	RootPackageGraphId int
	PackageGraphs []PackageGraph
	Packages []PackageInfo
	PackageTargetDirectories map[int]map[int]string
}

type LoadBuildGraphResult struct {
	IsSuccess bool
	Message *string
	Graph LoadBuildGraph
}